lib/timeline/timeline.dart

```diff
@@ -80,32 +87,48 @@ class TimelineScreen extends Screen {
         ..add(frameDetailsUI = new FrameDetailsUI()..attribute('hidden')),
     ]);
 
-    debugDrawButton.click(() {
+    trackWidgetBuildsButton.click(() {
       final bool wasSelected =
-          debugDrawButton.element.classes.contains('selected');
-      debugDrawButton.toggleClass('selected');
-      serviceInfo.service.callServiceExtension('ext.flutter.debugPaint',
-          isolateId: serviceInfo.isolateManager.selectedIsolate.id,
-          args: <String, bool>{'enabled': !wasSelected});
+          trackWidgetBuildsButton.element.classes.contains('selected');
+      trackWidgetBuildsButton.toggleClass('selected');
+      serviceInfo.service.callServiceExtension(
+        'ext.flutter.debugProfileBuilds',
+        isolateId: serviceInfo.isolateManager.selectedIsolate.id,
+        args: <String, bool>{'enabled': !wasSelected},
+      );
     });
 
     perfOverlayButton.click(() {
       final bool wasSelected =
           perfOverlayButton.element.classes.contains('selected');
       perfOverlayButton.toggleClass('selected');
       serviceInfo.service.callServiceExtension(
-          'ext.flutter.showPerformanceOverlay',
-          isolateId: serviceInfo.isolateManager.selectedIsolate.id,
-          args: <String, bool>{'enabled': !wasSelected});
+        'ext.flutter.showPerformanceOverlay',
+        isolateId: serviceInfo.isolateManager.selectedIsolate.id,
+        args: <String, bool>{'enabled': !wasSelected},
+      );
     });
 
-    debugBannerButton.click(() {
+    repaintRainbowButton.click(() {
       final bool wasSelected =
-          debugBannerButton.element.classes.contains('selected');
-      debugBannerButton.toggleClass('selected');
-      serviceInfo.service.callServiceExtension('ext.flutter.debugAllowBanner',
-          isolateId: serviceInfo.isolateManager.selectedIsolate.id,
-          args: <String, bool>{'enabled': !wasSelected});
+          repaintRainbowButton.element.classes.contains('selected');
+      repaintRainbowButton.toggleClass('selected');
+      serviceInfo.service.callServiceExtension(
+        'ext.flutter.repaintRainbow',
+        isolateId: serviceInfo.isolateManager.selectedIsolate.id,
+        args: <String, bool>{'enabled': !wasSelected},
+      );
+    });
+
+    debugDrawButton.click(() {
```

Comment:
Maybe worth pulling this implementation into a local function that takes the button element/service extension method name? As far as I can tell, they're all the same 8 lines the same except for those two values?

Reply:
Makes sense, will do.